src/index.js

```diff
@@ -99,9 +99,12 @@ export default async function fetch(url, options_) {
 			finalize();
 		});
 
-		fixResponseChunkedTransferBadEnding(request_, error => {
-			response.body.destroy(error);
-		});
+		if (!request_.shouldKeepAlive) {
```

Comment:
ping @tekwiz

also what do you think about this PR #1474

Reply:
@jimmywarting

`fixResponseChunkedTransferBadEnding` has the incorrect code (see my comments), https://github.com/node-fetch/node-fetch/pull/1474 should be accepted in any case.

---

About this #1325 pull request:

In case of `keep-alive` Agent `fixResponseChunkedTransferBadEnding` is not needed? Otherwise this pull request looks like just the problem ignoring by passing some property with breaking the code (if `fixResponseChunkedTransferBadEnding` is really needed function).

Reply:
`shouldKeepAlive` is a native property of the request object.
Although, there is no note about it in the doc, but it is.
It looks to be OK approach to detect is request uses keep-alive Agent, or no.